packages/compass-query-bar/src/stores/ai-query-reducer.ts

```diff
@@ -225,11 +237,11 @@ export const runAIQuery = (
       return;
     }
 
+    let query: Record<string, unknown>;
     let generatedFields: QueryFormFields;
     try {
-      generatedFields = parseQueryAttributesToFormFields(
-        jsonResponse.content.query
-      );
+      query = jsonResponse?.content?.query as Record<string, unknown>;
```

Comment:
This type assertion is incorrect, we validate the response before returning it from `getQuery` method, so what you get here will always match expected backend schema (or request will fail already and you will not end up in this part of the action)

Reply:
Yep gone after rebasing from main and refactoring.